### PR TITLE
Expose Vapor Request's request-id value, so that it can be passed into logging from other libraries

### PR DESCRIPTION
**Pull request by Joannis**
Exposes the Request's ID as a new property on `Request` as `requestID`. This is useful for any libraries or users who want to be able to trace the request ID through a request handler.

### CODE REVIEW

**Comment by Joannis** [comment]
See https://github.com/orlandos-nl/MongoKitten/issues/300

**Review by 0xTim — APPROVED** [review]
Thanks! This should get reused when it comes to tackling tracing as well

**Review by 0xTim — CHANGES_REQUESTED** [review]
Actually I'm going to be a real pain here - can you just add a simple test to make 2 requests and ensure the IDs are different? That'll protect us in the future for any changes and stop silly mistakes

File: Sources/Vapor/HTTP/Server/HTTPServerResponseEncoder.swift [hunk 1]
```diff
@@ -134,7 +134,7 @@ private final class ChannelResponseBodyStream: BodyStreamWriter {
             self.context.fireUserInboundEventTriggered(HTTPServerResponseEncoder.ResponseEndSentEvent())
             self.context.writeAndFlush(self.handler.wrapOutboundOut(.end(nil)), promise: promise)
             self.promise?.fail(error)
-        }
+            }
```
Comment (0xTim) [inline Sources/Vapor/HTTP/Server/HTTPServerResponseEncoder.swift:137]: Indentation
```suggestion
        }
```

**Comment by 0xTim** [comment]
Original comment:
> Libraries such as Fluent, MySQL, Postgres, MongoKitten & Redis currently have their own logging set up using swift-log. However, it can be useful to see the request-id from which a query was launched. Therefore I propose exposing the request-id as a property on `Request` so that end-users can set the metadata in other libraries' loggers as well. Therefore improving the debugability using these logs.

**Pull request merged**